src/Functions/URL/netloc.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "FunctionsURL.h"
+#include <common/find_symbols.h>
+
+
+namespace DB
+{
+
+struct ExtractNetloc
+{
+    /// We use the same as domain function
+    static size_t getReserveLengthForElement() { return 15; }
+
+    static void execute(Pos data, size_t size, Pos & res_data, size_t & res_size)
+    {
+        res_data = data;
+        res_size = size;
+
+        Pos pos = data;
+        Pos end = pos + size;
+
+        pos = find_first_symbols<'/'>(pos, end);
+        if (end == pos)
+            return;
+
+        /// Strings are zero-terminated.
```

Comment:
FixedString is not zero-terminated, so `pos[1]` can possibly be the first character of next url.